indexer/src/test/java/uk/ac/ebi/uniprot/indexer/uniprotkb/processor/UniProtEntryConverterTest.java

```diff
@@ -828,12 +870,83 @@ void testSubcellularLocationCommentConvertProperlyToDocument() throws Exception
+        suggestions.put(key,
+                        SuggestDocument.builder()
+                                .id(id)
+                                .value(homoSapiensFromDefaults)
+                                .build());
+
+        TaxonomicNode taxonomicNode = mock(TaxonomicNode.class);
+        when(taxonomicNode.id()).thenReturn(9606);
+        when(taxonomicNode.scientificName()).thenReturn(homoSapiensFromDBFile);
+        when(repoMock.retrieveNodeUsingTaxID(anyInt())).thenReturn(Optional.of(taxonomicNode));
+
+        assertThat(suggestions.entrySet(), hasSize(1));
+        SuggestDocument doc = suggestions.get(key);
+        assertThat(doc.altValues, is(empty()));
+        converter.setLineageTaxon(9606, new UniProtDocument());
+
+        assertThat(suggestions.entrySet(), hasSize(1));
+        assertThat(doc.altValues, is(empty()));
+        assertThat(doc.value, is(homoSapiensFromDBFile));
+    }
+
+    @Test
+    void taxonSuggestionAddedWhenAlreadyExistBefore() {
+        String id = "9606";
+        String homoSapiens = "Homo sapiens";
+
+        String key = converter.createSuggestionMapKey(SuggestDictionary.TAXONOMY, id);
+        suggestions.put(key,
+                        SuggestDocument.builder()
+                                .id(id)
+                                .value(homoSapiens)
+                                .build());
+
+        TaxonomicNode taxonomicNode = mock(TaxonomicNode.class);
+        when(taxonomicNode.id()).thenReturn(9606);
+        when(taxonomicNode.scientificName()).thenReturn(homoSapiens);
+        String synonym = "some synonym for homo sapiens";
+        when(taxonomicNode.commonName()).thenReturn(synonym);
+        when(repoMock.retrieveNodeUsingTaxID(anyInt())).thenReturn(Optional.of(taxonomicNode));
+
+        assertThat(suggestions.entrySet(), hasSize(1));
+        SuggestDocument doc = suggestions.get(key);
+        assertThat(doc.altValues, is(empty()));
+        converter.setLineageTaxon(9606, new UniProtDocument());
+
+        assertThat(suggestions.entrySet(), hasSize(1));
+        assertThat(doc.altValues, contains(synonym));
+    }
+
     private UniProtEntry parse(String file) throws Exception {
-        InputStream is = UniProtEntryDocumentPairProcessor.class.getClassLoader().getResourceAsStream("uniprotkb/" + file);
+        InputStream is = UniProtEntryDocumentPairProcessor.class.getClassLoader()
+                .getResourceAsStream("uniprotkb/" + file);
         assertNotNull(is);
         SupportingDataMap supportingDataMap = new SupportingDataMapImpl("uniprotkb/keywlist.txt",
                                                                         "uniprotkb/humdisease.txt",
-                                                                        "uniprotkb/PMID.GO.dr_ext.txt",
+                                                                        "target/test-classes/uniprotkb/PMID.GO.dr_ext.txt",
```

Comment:
Does it work with "uniprotkb/PMID.GO.dr_ext.txt"?

Reply:
yes